mijia/src/bluetooth/mod.rs

```diff
@@ -406,6 +511,34 @@ impl BluetoothSession {
         Ok(characteristics)
     }
 
+    /// Get a list of all descriptors on the given GATT characteristic.
+    pub async fn get_descriptors(
+        &self,
+        characteristic: &CharacteristicId,
+    ) -> Result<Vec<DescriptorInfo>, BluetoothError> {
+        let characteristic_node = self
+            .characteristic(characteristic)
+            .introspect_parse()
+            .compat()
+            .await?;
+        let mut descriptors = vec![];
+        for subnode in characteristic_node.nodes {
+            let subnode_name = subnode.name.as_ref().unwrap();
+            if subnode_name.starts_with("desc") {
```

Comment:
is this rule documented anywhere or should we *strictly* be introspecting each path and only returning things that have the appropriate dbus interface on them?

(I'm fine with this staying as is, but it probably wants a comment on it)

Reply:
It is documented incorrectly in https://git.kernel.org/pub/scm/bluetooth/bluez.git/tree/doc/gatt-api.txt that descriptors should have paths of the form `[variable prefix]/{hci0,hci1,...}/dev_XX_XX_XX_XX_XX_XX/serviceXX/charYYYY/descriptorZZZ`. The last part of the path is actually of the form `serviceXXXX/charYYYY/descZZZZ`.

Added a comment.

Reply:
An opportunity for your first bluez patch ;-)